tests/ui/models/filter_alerts_with_videos_test.js

```diff
@@ -0,0 +1,103 @@
+import testAlertSummaryWithVideos from '../mock/alerts_with_videos/alert_summary_with_browsertime_videos';
+import testAlertSummaryWithoutVideos from '../mock/alerts_with_videos/alert_summary_without_browsertime_videos';
+import currentJoblistWithVideoResultsOne from '../mock/alerts_with_videos/current_joblist_with_video_results_page_1';
+import currentJoblistWithVideoResultsTwo from '../mock/alerts_with_videos/current_joblist_with_video_results_page_2';
+import prevJoblistWithVideoResultsOne from '../mock/alerts_with_videos/prev_joblist_with_video_results_page_1';
+import prevJoblistWithVideoResultsTwo from '../mock/alerts_with_videos/prev_joblist_with_video_results_page_2';
+import currentJoblistWithoutVideoResultsOne from '../mock/alerts_with_videos/current_joblist_without_video_results_page_1';
+import currentJoblistWithoutVideoResultsTwo from '../mock/alerts_with_videos/current_joblist_without_video_results_page_2';
+import prevJoblistWithoutVideoResultsOne from '../mock/alerts_with_videos/prev_joblist_without_video_results_page_1';
+import prevJoblistWithoutVideoResultsTwo from '../mock/alerts_with_videos/prev_joblist_without_video_results_page_2';
+
+const dummyFrameworkName = 'browsertime';
+
+describe('FilterAlertsWithVideos', () => {
+  afterEach(() => {
+    fetchMock.reset();
+  });
+
+  describe('File bug url when alerts contain browsertime results links', () => {
+    beforeEach(() => {
+      fetchMock.mock(
+        `/api/jobs/?repo=autoland&push_id=847217`,
+        currentJoblistWithVideoResultsOne,
+      );
+      fetchMock.mock(
+        `/api/jobs/?repo=autoland&push_id=847217&page=2`,
+        currentJoblistWithVideoResultsTwo,
+      );
+      fetchMock.mock(
+        `/api/jobs/?repo=autoland&push_id=846998`,
+        prevJoblistWithVideoResultsOne,
+      );
+      fetchMock.mock(
+        `/api/jobs/?repo=autoland&push_id=846998&page=2`,
+        prevJoblistWithVideoResultsTwo,
+      );
+    });
+
+    test('should return alerts with browsertime results links', async () => {
+      const alertsWithVideos = new FilterAlertsWithVideos(
+        testAlertSummaryWithVideos,
+        [{ id: 13, name: dummyFrameworkName }],
+      );
+      const expectedResultsLink =
+        'https://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/c6nAPFniThOiaGvDT2DNCw/runs/0/artifacts/public/test_info/browsertime-results.tgz';
+      const expectedPrevResultsLink =
+        'https://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/LUjQR22tRp6J4wXLHhYk8g/runs/0/artifacts/public/test_info/browsertime-results.tgz';
+
+      const alerts = await alertsWithVideos.getFilteredAlerts();
+      alerts.forEach((alert) => {
+        if (alertsWithVideos.shouldHaveVideoLinks(alert)) {
+          expect(alert).toHaveProperty('results_link', expectedResultsLink);
+          expect(alert).toHaveProperty(
+            'prev_results_link',
+            expectedPrevResultsLink,
+          );
+        } else {
```

Comment:
We'll never reach this branch, because all our alerts should already have videos.

Reply:
only the ones that respect the criteria have video links.

Reply:
oh, you're right.